### PR TITLE
Log as much errors with vault and consul

### PR DESCRIPTION
**Pull request by adelowo**
## Description
Log errors if any when fetching keys from consul and value

## Related Issue

On a support call with a customer and we cannot retrieve data from Vault. Everything else seems right and i 
cannot find any more error per why the keys cannot be retrieved. 

## How This Has Been Tested

Manually :) 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [x] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

File: gateway/server.go [hunk 1]
```diff
@@ -1120,6 +1120,7 @@ func kvStore(value string) (string, error) {
 		key := strings.TrimPrefix(value, "consul://")
 		log.Debugf("Retrieving %s from consul", key)
 		if err := setUpConsul(); err != nil {
+			log.WithError(err).Error(err)
```
Comment (gernest) [inline gateway/server.go:1123]:
```suggestion
			log.Error("Failed to setup consul: ", err)
```

File: gateway/server.go [hunk 2]
```diff
@@ -1131,6 +1132,7 @@ func kvStore(value string) (string, error) {
 		key := strings.TrimPrefix(value, "vault://")
 		log.Debugf("Retrieving %s from vault", key)
 		if err := setUpVault(); err != nil {
+			log.WithError(err).Error(err)
```
Comment (gernest) [inline gateway/server.go:1135]:
```suggestion
			log.Error("Failed to setup vault: ", err)
```

**Review by gernest — CHANGES_REQUESTED** [review]
I left a few suggestion. It is just a nit pick, its nice to be consistent when logging stuff with the rest of the gw.

**Review by gernest — APPROVED** [review]
LGTM

**Review by furkansenharputlu — APPROVED** [review]

**Pull request merged**